pkg/kubelet/kuberuntime/kuberuntime_gc.go

```diff
@@ -279,31 +298,26 @@ func (cgc *containerGC) evictSandboxes() error {
 			}
 		}
 		if hasContainers {
-			continue
+			sandboxInfo.active = true
 		}
 
-		podUID := types.UID(sandbox.Metadata.Uid)
-		evictSandboxes[podUID] = append(evictSandboxes[podUID], sandboxGCInfo{
-			id:         sandbox.Id,
-			createTime: time.Unix(0, sandbox.CreatedAt),
-		})
+		sandboxesByPod[podUID] = append(sandboxesByPod[podUID], sandboxInfo)
 	}
 
 	// Sort the sandboxes by age.
-	for uid := range evictSandboxes {
-		sort.Sort(sandboxByCreated(evictSandboxes[uid]))
+	for uid := range sandboxesByPod {
+		sort.Sort(sandboxByCreated(sandboxesByPod[uid]))
 	}
 
-	for podUID, sandboxes := range evictSandboxes {
+	for podUID, sandboxes := range sandboxesByPod {
 		if cgc.isPodDeleted(podUID) {
-			// Remove all sandboxes if pod has been deleted.
+			// Remove all evictable sandboxes if the pod has been removed.
```

Comment:
> In you comment, you said that "keep the latest one (regardless of its state)", but here I didn't see that.

Sorry, it is in the previous comment. I mean the latest one should be kept, whether it is ready or not. And other active ones are also kept. Will make an update to the comment:

* remove all inactive sandboxes if pod has been removed
* else
  * keep active sandboxes
  * keep the latest sandbox whether it is active or not
  * remove other inactive sandboxes